### ISSUE TITLE
Ensure we don't send customer messages back to the customer. seems to be handled incorrectly

### ISSUE THREAD

**Issue by intptr-t** [amazon-connect/amazon-connect-message-streaming-examples#7]
### 1. CUSTOMER and ALL are undefined
In the following process, `CUSTOMER` and `ALL` are undefined.

https://github.com/amazon-connect/amazon-connect-message-streaming-examples/blob/f5f8712e581ccc44ee5c6d40ea5df2874cb506c0/src/lambda/outboundMessageHandler/index.js#L72-L73

### Cloud Watch Logs
```log
ERROR	Invoke Error 	{
    "errorType": "ReferenceError",
    "errorMessage": "CUSTOMER is not defined",
...
}
```

### 2. SNS payload type error

I have checked the [Example SNS payload](https://docs.aws.amazon.com/connect/latest/adminguide/sns-payload.html#sns-message-payload) and it seems that `record.Sns.MessageAttributes.MessageVisibility` is not correct and
`record.Sns.MessageAttributes.MessageVisibility.Value` is correct.

### 3. I feel uncomfortable with the conditional expression. L72-L74

The following equation is

https://github.com/amazon-connect/amazon-connect-message-streaming-examples/blob/f5f8712e581ccc44ee5c6d40ea5df2874cb506c0/src/lambda/outboundMessageHandler/index.js#L72-L74

equivalent to

```javascript
 (record.Sns.MessageAttributes.MessageVisibility != CUSTOMER &&
  record.Sns.MessageAttributes.MessageVisibility == ALL)
 ```

Is the conditional expression correct?

**Comment by ahmedyounis** [comment]
Any update on this issue? Its an issue that pretty much makes the whole solutions provided by the solution AWS useless. What good is a two way sms communication if the customer cant see what the agent is writing? 

https://aws.amazon.com/blogs/contact-center/building-personalized-customer-experiences-over-sms-through-amazon-connect/

**Comment by intptr-t** [comment]
Nos. 1-2 have been resolved by #13. (Thanks!)

No.3

The ["Amazon SNS payload used in message streaming"](https://docs.aws.amazon.com/connect/latest/adminguide/sns-
payload.html) document now clearly indicates the value range for MessageVisibility.
> MessageVisibility: Possible values: AGENT, CUSTOMER, ALL.

Following that, the conclusion of the No.3 uncomfortable seems to be just looking at ALL.


**Logic1**

```javascript
((record.Sns.MessageAttributes.MessageVisibility == CUSTOMER ||  
  record.Sns.MessageAttributes.MessageVisibility == ALL) &&  
  record.Sns.MessageAttributes.MessageVisibility != CUSTOMER) 
```


MessageVisibility        | Logic1
-------------------------|-----------------------------------|
ALL                      | (false \|\| true) && true => true   |
CUSTOMER                 | (true \|\| false) && false => false |
AGENT                    | (false \|\| false) && true => false |

I don't mind as it seems to be working for me at least :p
